From 74ea60ab0d1aeaa3f5018e99b2729d2edbdf0a9e Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Wed, 3 Sep 2025 21:23:06 +0300
Subject: [PATCH 3/4] pci: layerscape_gen4: add missing <linux/delay.h> header
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes the following build error:

drivers/pci/pcie_layerscape_gen4.c: In function ‘ls_pcie_g4_wait_for_link’:
drivers/pci/pcie_layerscape_gen4.c:66:17: error: implicit declaration of function ‘udelay’ [-Wimplicit-function-declaration]
   66 |                 udelay(LINK_WAIT_TIMEOUT);
      |                 ^~~~~~

To be squashed into commit "pci: ls_pcie_g4: Wait 100ms for Link Up in
ls_pcie_g4_probe".

Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 drivers/pci/pcie_layerscape_gen4.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pci/pcie_layerscape_gen4.c b/drivers/pci/pcie_layerscape_gen4.c
index 2fbb507c628e..c6b823b3a661 100644
--- a/drivers/pci/pcie_layerscape_gen4.c
+++ b/drivers/pci/pcie_layerscape_gen4.c
@@ -15,6 +15,7 @@
 #include <errno.h>
 #include <malloc.h>
 #include <dm.h>
+#include <linux/delay.h>
 #include <linux/sizes.h>
 
 #include "pcie_layerscape_gen4.h"
-- 
2.43.0

